### PR TITLE
Add channels concept

### PR DESCRIPTION
**Pull request by RezaSi**
for this https://github.com/exercism/go/issues/2182

### CODE REVIEW

**Review by junedev — CHANGES_REQUESTED** [review]
The content you present here seems to be copied from https://www.golangbootcamp.com/book/concurrency. The book does not have any license file so we cannot know if the author allows copying without attribution. https://github.com/GoBootcamp/book/issues/48 This is too risky for us.

The content needs to written in your own words, copied from places that explicitly allow copying without attribution (e.g. via an MIT license) or the sentences that are copied need to be quoted and the source needs to be added as a footnote (this is legally only allowed for short sections if not explicitly permitted otherwise).

I only looked at the first part so far. I don't know for how much of the total content this applies to. It's easier for you to know. In any case, please rework all the copied content so it follows the rules above. If you have any questions, let me know.

**Comment by junedev** [comment]
Did you invent the exercise from scratch? If yes, I could start reviewing that part.

**Comment by RezaSi** [comment]
> The content you present here seems to be copied from https://www.golangbootcamp.com/book/concurrency. The book does not have any license file so we cannot know if the author allows copying without attribution. [GoBootcamp/book#48](https://github.com/GoBootcamp/book/issues/48) This is too risky for us.
> 
> The content needs to written in your own words, copied from places that explicitly allow copying without attribution (e.g. via an MIT license) or the sentences that are copied need to be quoted and the source needs to be added as a footnote (this is legally only allowed for short sections if not explicitly permitted otherwise).
> 
> I only looked at the first part so far. I don't know for how much of the total content this applies to. It's easier for you to know. In any case, please rework all the copied content so it follows the rules above. If you have any questions, let me know.

Yes, some part of the intro and about is copied from this site and golang tut, I don't know this, but it's ok, I will try to write it again with my words!

**Comment by RezaSi** [comment]
> Did you invent the exercise from scratch? If yes, I could start reviewing that part.

Yes, it's not copied from somewhere or another exercise, tnx for the review.

**Review by junedev — CHANGES_REQUESTED** [review]
I had a look at the about.md file. I will check out the exercise later on.

**Comment by RezaSi** [comment]
@junedev Tnx for all your advice, I think resolved all, and repair some other parts.
Yes, I'm not a native speaker, if you can help with grammar, it's very welcome.

**Comment by RezaSi** [comment]
@junedev Tnx for all your advice, I think resolved all, and repair some other parts.
Yes, I'm not a native speaker, if you can help with grammar, it's very welcome.

**Comment by RezaSi** [comment]
@junedev it's blocked? :-?

**Comment by junedev** [comment]
@RezaSi Sorry for the wait. I had to discuss with the Exercism leadership how to proceed with this PR. The problem is that the concept content still contains some sentences that are copied without attribution and some others were there were only slight changes made to the original content. This happened  after I mentioned the issue twice already. It is too risky for us to merge content like this and it takes too much time for me to check and discuss every sentence. I will still consider the exercise part (without introduction.md) for merging but before I merge, I will remove the concept content from the PR and set the exercise status to "work in progress". Then there will be a follow up issue about writing the concept from scratch and once someone else worked on this, the exercise can go live.

Hopefully, I can find some time to review the exercise on the weekend but I can't promise. I'm sorry this all takes so long but keep in mind that all track maintainers on Exercism are volunteers that do this work in their free time.

File: exercises/concept/cache/.meta/example.go
Comment (junedev) [inline exercises/concept/cache/.meta/example.go:36]: A couple of thoughts about this task (and the solution):
* Currently, the function always returns nil so it could also just not have a return value at all.
* The instructions should clarify what happens in case Set returns an error.
* Overall, the required solution here is very similar to the one of the first task. Usually we try to bring up some new aspect in each task. Have another look at the learning objectives, maybe the task can be adjusted/re-written to cover something different. For example, currently your tasks don't cover ranging over a channel at all.

File: exercises/concept/cache/cache.go
Comment (junedev) [inline exercises/concept/cache/cache.go:3]: Please remove the names from the return values. Those should only be used when really needed as they can lead to mistakes easily.

File: exercises/concept/cache/cache_struct.go
Comment (junedev) [inline exercises/concept/cache/cache_struct.go:1]: If you follow the suggestion mentioned above with the interface, the cache implementation can be part of the standard test file. If you want to keep it as a separate file, you can list it in the config under the "editor" key so it shows up when solving the exercise but cannot be edited on the website.

Comment (junedev) [inline exercises/concept/cache/cache_struct.go:1]: Instead of "cache_struct.go", "types.go" is a more common name for a file like this.

**Review by junedev — CHANGES_REQUESTED** [review]
As mentioned before, I reviewed the exercise. I like the general idea about fetching data concurrently. I did not review the test file yet as it might change as you make some other adjustments. I will take a closer look once the other comments are addressed.

**Comment by junedev** [comment]
Closing this as it seems abandoned.

**Pull request closed without merging**
